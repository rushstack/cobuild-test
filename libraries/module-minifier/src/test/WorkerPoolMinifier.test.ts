// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
// See LICENSE in the project root for license information.

import type { IMinifierConnection } from '../types';

let terserVersion: string = '1.0.0';
jest.mock('terser/package.json', () => {
  return {
    get version(): string {
      return terserVersion;
    }
  };
});

describe('WorkerPoolMinifier', () => {
  it('Includes terserOptions in config hash', async () => {
    const { WorkerPoolMinifier } = await import('../WorkerPoolMinifier');
    type WorkerPoolMinifier = typeof WorkerPoolMinifier.prototype;

    const minifier1: WorkerPoolMinifier = new WorkerPoolMinifier({
      terserOptions: {
        ecma: 5
      }
    });
    const minifier2: WorkerPoolMinifier = new WorkerPoolMinifier({
      terserOptions: {
        ecma: 2015
      }
    });

    const connection1: IMinifierConnection = await minifier1.connect();
    await connection1.disconnect();
    const connection2: IMinifierConnection = await minifier2.connect();
    await connection2.disconnect();

    expect(connection1.configHash).toMatchSnapshot('ecma5');
    expect(connection2.configHash).toMatchSnapshot('ecma2015');
    expect(connection1.configHash !== connection2.configHash);
  });

  it('Includes terser package version in config hash', async () => {
    const { WorkerPoolMinifier } = await import('../WorkerPoolMinifier');
    type WorkerPoolMinifier = typeof WorkerPoolMinifier.prototype;

    terserVersion = '5.9.1';
    const minifier1: WorkerPoolMinifier = new WorkerPoolMinifier({});
    terserVersion = '5.16.2';
    const minifier2: WorkerPoolMinifier = new WorkerPoolMinifier({});

    const connection1: IMinifierConnection = await minifier1.connect();
    await connection1.disconnect();
    const connection2: IMinifierConnection = await minifier2.connect();
    await connection2.disconnect();

    expect(connection1.configHash).toMatchSnapshot('terser-5.9.1');
    expect(connection2.configHash).toMatchSnapshot('terser-5.16.1');
    expect(connection1.configHash !== connection2.configHash);
  });
});
