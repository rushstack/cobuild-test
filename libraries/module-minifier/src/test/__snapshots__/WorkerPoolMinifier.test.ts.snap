// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkerPoolMinifier Includes terser package version in config hash: terser-5.9.1 1`] = `"0cjBnphY49XdxkmLxJLfNoRQbl3R6HdeT8WNl56oZWk="`;

exports[`WorkerPoolMinifier Includes terser package version in config hash: terser-5.16.1 1`] = `"dYinCy9XUHWWDXSIwq7F/GvAYLxxv3WjZdHIpd2sZpc="`;

exports[`WorkerPoolMinifier Includes terserOptions in config hash: ecma5 1`] = `"FcgklO/zYzwo0R2IM6OKH5uLA0hXVekjeqmFqKweKCg="`;

exports[`WorkerPoolMinifier Includes terserOptions in config hash: ecma2015 1`] = `"Vku2B3kuB95N1Xo2q8J/nAJrmxwXV+fQmmUwp4xj+Xk="`;
